spec/models/income_spec.rb

```diff
@@ -1,11 +1,22 @@
 require 'rails_helper'
 
 RSpec.describe Income, type: :model do
-  it { should have_db_column :amount}
+
+  describe 'DB table' do
+    it { should have_db_column :amount}
+  end
+
+  describe 'Associations' do
+    it { should belong_to :user}
+  end
+
+  describe 'Validations' do
+    it {validate_presence_of :amount}
+  end
 
   describe 'Factory' do
     it 'should have valid Factory' do
-      expect(FactoryBot.create(:income)).to be_valid
+      expect(FactoryBot.create(:user)).to be_valid
```

Comment:
same as the expense factory, you need to pass in a user to associate the income to user